server/routes/warningDetails.ts

```diff
@@ -96,7 +96,10 @@ export default function warningDetailsRoutes(
             requirementId: enforceableContactWithRequirement?.requirement?.id,
             requirementTypeMainCategoryDescription: enforceableContactWithRequirement?.requirement?.type?.description,
             requirementTypeSubCategoryDescription: enforceableContactWithRequirement?.requirement?.subType?.description,
-            rejectionReason: req.body[bodyParamBreachReason],
+            rejectionReason: warningDetails.breachReasons
+              .filter(c => c.code === req.body[bodyParamBreachReason])
+              .map(c => c.description)
+              .toString(),
```

Comment:
This should achieve the same result, but only return the first item if there are somehow multiple matching items in the list:
```suggestion
            rejectionReason: warningDetails.breachReasons.find(c => c.code === req.body[bodyParamBreachReason]).description,
```